x/distribution/types/validator.go

```diff
@@ -8,9 +8,25 @@ import (
 )
 
 // historical rewards for a validator
-// TODO add reference counter, ref https://github.com/cosmos/cosmos-sdk/pull/3099#discussion_r245747051
 // height is implicit within the store key
-type ValidatorHistoricalRewards = sdk.DecCoins
+// the reference count indicates the number of objects
+// which might need to reference this historical entry
+// at any point, it is equal to the number of outstanding
+// delegations starting on the associated period, plus
+// the number of slashes at the associated period, plus one
+// per validator for the zeroeth period
```

Comment:
This sentence is squeezing in a lot. I think it makes sense to break this comment down a bit, maybe more pseudo-code. Here is a proposed change

```
// The reference count indicates the number of objects
// which might need to reference this historical entry
// at any point.
// ReferenceCount = 
//     number of outstanding delegations (starting on the associated period)
//     + number of slashes at the associated period
//     + (only if zeroeth period) one per validator
```

Within this, I'm confused about what `number of outstanding delegations starting on the associated period` actually is, specifically the word "starting". Maybe there is a more explicit way of wording this. For instance, is the number of delegations which were made during this period? does this include what is "beyond" this period because this is only "starting" at this period? ... more clarity would be appreciated 
_________________
Additionally, a comment on what `CumulativeRewardRatio` is would be good

Reply:
Clarified, added a comment on `CumulativeRewardRatio`.